components/cdc/src/delegate.rs

```diff
@@ -780,16 +760,20 @@ impl Delegate {
                     }
                 }
 
-                *occupied = row;
+                let occupied = rows.entry(row.key.clone()).or_default();
+                if occupied.1 {
+                    assert!(!has_value);
+                    has_value = true;
+                    mem::swap(&mut occupied.0.value, &mut row.value);
+                }
+                *occupied = (row, has_value);
```

Comment:
How about `has_value = !row.value.is_empty()`? So we don't have to change `decode_write` and `decode_default`.

Reply:
This makes sense to me.

Reply:
+1 👍

Reply:
We have discussed it offline. If there is an empty short value, `has_value` should also be `true` but `row.value` will be empty.